common/src/stack/command/stack/argument_processors/repo.py

```diff
@@ -0,0 +1,178 @@
+from collections import namedtuple
+
+from stack.exception import ArgNotFound
+
+class RepoArgProcessor:
+	REQUIRED_REPO_COLUMNS = [
+		'name',
+		'alias',
+		'url',
+	]
+
+	OPTIONAL_REPO_COLUMNS = [
+		'autorefresh',
+		'assumeyes',
+		'type',
+		'is_mirrorlist',
+		'gpgcheck',
+		'gpgkey',
+		'os',
+		'pallet_id',
+	]
+
+	REPO_COLUMNS = REQUIRED_REPO_COLUMNS + OPTIONAL_REPO_COLUMNS
+
+	def insert_repo(self, name, alias, url, **kwargs):
+		'''
+		Insert a repo with optional kwargs into the repos table.
+		This works like add pallet - if you try to add a repo that already exists, silently succeed
+		'''
+		data = {}
+		for key in self.OPTIONAL_REPO_COLUMNS:
+			if key in kwargs:
+				data[key] = kwargs[key]
+
+		where_clause = ' AND '.join(f'{col}=%s' for col in self.REQUIRED_REPO_COLUMNS + list(data))
+		select_stmt = f'''id FROM repos WHERE {where_clause}'''
+
+		if len(self.db.select(select_stmt, (name, alias, url) + tuple(data.values()))) != 0:
+			return
+
+		cols = ', '.join(['name', 'alias', 'url'] + list(data))
+		vals = ', '.join(['%s'] * (3 + len(data.values())))
+
+		sql = f'''INSERT INTO repos
+			({cols})
+			VALUES
+			({vals})
+			'''
+
+		self.db.execute(sql, (name, alias, url, *data.values()))
+
+	def delete_repo(self, repo):
+		''' delete the repo from the database, should also remove any box associations '''
+		sql = '''DELETE FROM repos WHERE name LIKE BINARY %s OR alias LIKE BINARY %s '''
```

Comment:
Why are we treating repo names and aliases as case-sensitive, when you have a unique constraint on the fields, which isn't case-sensitive. So, the DB would never let you have a repo name or alias that was a duplicate in a case-insensitive manner.

Reply:
Got rid of all the Binary's.